tasks/main.yml

```diff
@@ -126,7 +126,22 @@
   copy:
     content: "{{ humio_permissions | default('{}') }}"
     dest: "{{ humio_data_path }}/{{ humio_host_id }}-{{ item.key }}/data/view-group-permissions.json"
+    mode: 0644
+    owner: humio
+    group: humio
+  with_dict: "{{ ansible_local.cpusockets }}"
+  when: humio_permissions_path is undefined
+
+- name: Copy permissions
+  tags: humio-permissions
+  copy:
+    src: "{{ humio_permissions_path }}"
```

Comment:
In other project we have the exact same problem with large files, therefore we use Ansible's [`lookup`](https://docs.ansible.com/ansible/latest/plugins/lookup.html), i.e.
```
humio_permissions: "{{ lookup('file', 'permissions.json') }}"
```
Unless there's a very good reason for not using that, I prefer to not complicate the role further.

Reply:
I decided to just back this change out for now to not complicate the permissions issue. I did spend some time looking at this and there isn't a super great way of doing this simply while maintaining backwards compatibility. We can tackle it in another PR later.